File: DSA/fundamentals/arrays_and_lists/part2/java/Judge.java
package DSA.fundamentals.arrays_and_lists.part2.java;

import java.util.*;

public class Judge {

    public void runTests() {
        runTests(
                new String[][]{
                    {"Plateau", "Haven", "Insomnia", "Grand Trad", "Plateau", "Sunville"},
                    {"Plateau", "Haven", "Insomnia", "Grand Trad", "Sunville", "Bronze City", "Sunville"},}
        );
    }

    public void runTests(String[][] testCasesC) {
        int passes = 0;
        for (int i = 0; i < testCasesC.length; i++) {
            if (test(testCasesC[i], testCaseSolutionOutput(testCasesC[i]))) {
                passes++;
            }
        }
        System.out.println("Passed " + passes + " of " + testCasesC.length + " tests.");
    }

    private boolean test(String[] c, String expected) {
        Solution sol = new Solution();
        String output = sol.dispatchTo(c);
        boolean passed = output.equals(expected);

        System.out.println("c: " + Arrays.toString(c));
        System.out.println("Expected Output: " + expected);
        System.out.println("Your Output: " + output);
        System.out.println(passed ? "Test PASSED" : "Test FAILED");
        System.out.println("-----------");
        return passed;
    }

    private String testCaseSolutionOutput(String[] c) {
        ArrayList<String> seen = new ArrayList<>();
        for (String city : c) {
            if (seen.contains(city)) {
                return city;
            }
            seen.add(city);
        }
        return "";
    }

}

File: DSA/fundamentals/arrays_and_lists/part2/python/judge.py
from solution import Solution

class Judge:
    def run_tests(self):
        test_cases_c = [
            ["Plateau", "Haven", "Insomnia", "Grand Trad", "Plateau", "Sunville"],
            ["Plateau", "Haven", "Insomnia", "Grand Trad", "Sunville", "Bronze City", "Sunville"],
        ]

        passes = 0;
        for i in range(len(test_cases_c)):
            if self.test(test_cases_c[i].copy(), self.test_case_solution_output(test_cases_c[i].copy())):
                passes += 1
        print(f"Passed {passes} of {len(test_cases_c)} tests.")

    def test(self, c, expected):
        solution = Solution()
        output = solution.dispatch_to(c)
        passed = output == expected

        print(f"c: {c}")
        print(f"Expected Output: {expected}")
        print(f"Your Output: {output}")
        print("Test " + ("PASSED" if passed else "FAILED"))
        print("-----------")
        return passed

    def test_case_solution_output(self, c):
        seen = []
        for city in c:
            if city in seen:
                return city
            seen.append(city)
        return ""